### ISSUE TITLE
Add aria labels that are missing in HTML report 

### ISSUE THREAD

**Issue by mitchelbaker-cisa** [cisagov/ScubaGear#972]
# 💡 Summary #

There's a few instances where adding aria labels will help to improve accessibility of the HTML reports. 

## Motivation and context ##

Relates to #879 

WCAG 2.2

[Guideline 1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships.html)

[Guideline 4.1.2 Name, Role, Value](
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value)

## Implementation notes ##

1. Light mode/dark mode is used to describe a checkbox input, add aria-labeledby attribute to better describe the relationship

```
<p id="toggle-text">Light Mode</p>
<label class="switch">
  <input id="toggle" type="checkbox" onclick="toggleDarkMode()" aria-labelledby="toggle-text">
  <span class="slider round"></span>
</label>
```

2. For "Expand all"/"Collapse all" buttons for the CAP table, a title attribute is used to display a tooltip. Add aria-label here to better describe the action for each button, e.g. "Expands all rows in the conditional access policy table"

```
<button title="Expand all" aria-label="Expands all rows in the conditional access policy table">+ Expand all</button>
```

## Acceptance criteria ##

How do we know when this work is done?

- [x] Elements in the HTML report should be given the attribute aria-label to provide an accessible name for situations when there is no visible label due to a chosen design approach
- [x] The aria-labelledby attribute should be used to associate a user interface control, such as a form field, with text on the page that labels it. 
- [ ] Smoke tests pass to indicate any added information displays correctly

**Comment by gdasher** [comment]
I think you're saying that we should add aria labels, which is good.  But the current reports do have text alternatives in both cases -- ie the table cells have the words "fail"/"pass" in them.  I believe that should come out okay when using a screen reader.  

Should we rename this to be about adding labels?  Do we know if that will make it better or worse?  My understanding is you use that for anonymous elements and my worry is it would cause repeated statements (one for the label and one for the actual text in the table)

**Comment by mitchelbaker-cisa** [comment]
> I think you're saying that we should add aria labels, which is good. But the current reports do have text alternatives in both cases -- ie the table cells have the words "fail"/"pass" in them. I believe that should come out okay when using a screen reader.
> 
> Should we rename this to be about adding labels? Do we know if that will make it better or worse? My understanding is you use that for anonymous elements and my worry is it would cause repeated statements (one for the label and one for the actual text in the table)

Good point, didn't take into account "pass"/"fail"/"warning"/"N/A" in the result column which solves the acceptance criteria of having a text alternative. https://stackoverflow.com/a/62824380 provides a similar example. "You can add anything you like, i.e. text or icon. It's fine as long as there is another way than just color to get the information."

Renaming to be about adding labels will be more specific. Another use case I'm seeing where a label may be relevant is for the CAP "expand all"/"collapse all" buttons. They have title attributes, which provides a tooltip, and the button's inner text would be read, but it seems a bit vague as to what function is being done here.
